### PR TITLE
Identity bugs fix July 2020

### LINKED ISSUE #12220: [TEST BUG] com.azure:azure-identity desktop builds fail on Windows due to locale issue.
**Describe the bug**
Whilst running the following command:

```
mvn test -pl com.azure:azure-identity -am
```

I noticed a failing test on my Windows 10 workstation for the MSITokenTests class. It was failing on the ```canParseDateTime12Hr``` test case. The underlying reason was locale. When calling ```parse(...)``` on the DateTimeFormatter it blew up when it go to the AM/PM.

Turns out that this is some kind of locale issue. If I swap my workstation to the US region the test passes.

***Exception or Stack Trace***

```
java.lang.IllegalArgumentException: Unable to parse date time 1/10/2020 3:03:28 PM +00:00
    at com.azure.identity@1.1.0-beta.6/com.azure.identity.implementation.MSITokenTests.canParseDateTime12Hr(MSITokenTests.java:29)
```

**To Reproduce**

Run the following command:

```
mvn test -pl com.azure:azure-identity -am
```

Then review the MSITokenTest sure fire reports to see the exception.

***Code Snippet***
Add the code snippet that causes the issue.

https://github.com/Azure/azure-sdk-for-java/blob/f4341a47bff66eeada66cf44310c6cd00df21dcf/sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java#L58

Assuming the locale returned from the MSI endpoint is always US, we could potentially append something like ```.withLocale(Locale.US)```.

**Expected behavior**

Test should pass regardless of machine locale.

**Screenshots**

**Setup (please complete the following information):**
 - OS: Windows 10 64-bit
 - IDE : Command-line
 - Version of the Library used: master branch

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### LINKED ISSUE #12275: [BUG] port 0 is default value in InteractiveBrowserCredentialBuilder
**Describe the bug**
The detailed info can be found [here](https://github.com/Azure/azure-libraries-for-java/issues/1219)

***Exception or Stack Trace***
`This site can’t be reachedThe webpage at http://localhost:0/?code=CODE&session_state=SESSION might be temporarily down or it may have moved permanently to a new web address.`

**To Reproduce**
Authenticate without setting port value or setting 0 there.

***Code Snippet***
`withPort(0)`

**Expected behavior**
If the value of port is `0`, the system will pick next available port. The expected might be to add validation part for value check.

**Screenshots**

**Setup (please complete the following information):**
 - OS: Windows
 - IDE : IntelliJ
 - Version of the Library used: beta.5
 - JDK: 1.8

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes #12220 
Fixes #12275

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * @see InteractiveBrowserCredential
  */
 public class InteractiveBrowserCredentialBuilder extends AadCredentialBuilderBase<InteractiveBrowserCredentialBuilder> {
-    private int port;
+    private Integer port;
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java:17]: Changes from primitive to class.
Primitive gives port a default value of zero.
This is a required parameter and its value should come from user input.

Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java:17]: It's a required value so the value would never be the default though?

Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java:17]: That's what my change fixed.
previously port was a primitive, and that had a default value of 0 and consequently builder would never complain in cases when user didn't supply the port as the default value would be zero.

Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java:17]: I see. Make sense

**Pull request merged**